app/portainer/views/endpoints/edit/endpoint.html

```diff
@@ -54,29 +54,44 @@
         </div>
       </div>
 
-      <por-switch-field
-        label="'Allow self-signed certs'"
-        checked="state.allowSelfSignedCerts"
-        tooltip="'When allowing self-signed certificates the edge agent will ignore the domain validation when connecting to Portainer via HTTPS'"
-        on-change="(onToggleAllowSelfSignedCerts)"
-      ></por-switch-field>
+      <div class="form-group">
+        <por-switch-field
+          label="'Allow self-signed certs'"
+          checked="state.allowSelfSignedCerts"
+          tooltip="'When allowing self-signed certificates the edge agent will ignore the domain validation when connecting to Portainer via HTTPS'"
+          on-change="(onToggleAllowSelfSignedCerts)"
+        ></por-switch-field>
+      </div>
+
+      <div class="form-group">
+        <label for="env_vars" class="col-sm-3 col-lg-2 control-label text-left" style="padding-left: 0; padding-top: 5px">
+          Environment variables
+          <portainer-tooltip
+            position="bottom"
+            message="Comma separated list of environment variables that will be sourced from the host where the agent is deployed."
+          ></portainer-tooltip>
+        </label>
+        <div class="col-sm-9 col-lg-10">
+          <input type="text" class="form-control" id="env_vars" ng-model="formValues.EnvVarSource" placeholder="var1,var2" />
```

Comment:
a small suggestion, if this allows the user to pass envs in a `var1=value1,var2` manner, I'd say to change the placeholder to something in that format